src/emulator/storage/files.ts

```diff
@@ -125,12 +156,72 @@ export class StorageLayer {
     private _cloudFunctions: StorageCloudFunctions
   ) {}
 
+  /**
+   * Create SignedUrl
+   * @date 6/28/2023 - 2:56:09 PM
+   *
+   * @async
+   * @param {GetObjectRequest} request
+   * @returns {Promise<boolean>}
+   */
+  async generateSignedUrl(request: CreateSignedUrl): Promise<SignedUrlResponse> {
+    const metadata = this.getMetadata(request.bucketId, request.decodedObjectId);
+
+    const currentDate = this.getCurrentDate();
+
+    const timeToLive = request.ttlSeconds;
+    if (!Number.isInteger(timeToLive)) {
+      throw new BadRequestError("Invalid TTL Parameter");
+    }
+
+    if (timeToLive < SIGNED_URL_MIN_TTL_SECONDS || timeToLive > SIGNED_URL_MAX_TTL_SECONDS) {
+      throw new BadRequestError(`TTL specified is less than 0 or more than allowed max (2 week)`);
+    }
+
+    const authorized = await this._rulesValidator.validate(
+      ["b", request.bucketId, "o", request.decodedObjectId].join("/"),
+      request.bucketId,
+      RulesetOperationMethod.GET,
+      { before: metadata?.asRulesResource() },
+      this._projectId,
+      request.authorization
+    );
+
+    if (!authorized) {
+      throw new ForbiddenError();
+    }
+
+    if (!metadata) {
+      throw new NotFoundError("Object in bucket does not exist");
+    }
+
+    const unsignedUrl = createUnsignedUrl({
```

Comment:
nit: I think having a helper for this is a bit overkill. We could just inline a variable:

```
const unsigned = `${url}/v0/b/${bucketId}/o/${encodeURIComponent(
    decodedObjectId
  )}?alt=media&X-Firebase-Date=${urlUsableSeconds}&X-Firebase-Expires=${urlTtlSeconds}`
```

That way someone reading this function can see the full structure of the url.

Reply:
the reason i have it as a helper is because i use this same block of code in my post reuqest, get request, and testing when creating unsignedUrls